app/src/main/java/org/mozilla/fenix/home/intent/StartSearchIntentProcessor.kt

```diff
@@ -7,41 +7,58 @@ package org.mozilla.fenix.home.intent
 import android.content.Intent
 import androidx.navigation.NavController
 import androidx.navigation.navOptions
+import mozilla.components.browser.state.store.BrowserStore
 import mozilla.components.service.glean.private.NoExtras
 import org.mozilla.fenix.GleanMetrics.SearchWidget
 import org.mozilla.fenix.HomeActivity
 import org.mozilla.fenix.NavGraphDirections
 import org.mozilla.fenix.R
+import org.mozilla.fenix.browser.waitForPIPToClose
 import org.mozilla.fenix.components.metrics.MetricController
 import org.mozilla.fenix.components.metrics.MetricsUtils
 import org.mozilla.fenix.ext.nav
+import org.mozilla.gecko.util.ThreadUtils.runOnUiThread
 
 /**
  * When the search widget is tapped, Fenix should open to the search fragment.
  * Tapping the private browsing mode launcher icon should also open to the search fragment.
  */
 class StartSearchIntentProcessor(
-    private val metrics: MetricController
+    private val metrics: MetricController,
+    private val store: BrowserStore
 ) : HomeIntentProcessor {
 
     override fun process(intent: Intent, navController: NavController, out: Intent): Boolean {
         val event = intent.extras?.getString(HomeActivity.OPEN_TO_SEARCH)
+        var processed = false
+
         return if (event != null) {
-            val source = when (event) {
-                SEARCH_WIDGET -> {
-                    SearchWidget.newTabButton.record(NoExtras())
-                    MetricsUtils.Source.WIDGET
-                }
-                STATIC_SHORTCUT_NEW_TAB,
-                STATIC_SHORTCUT_NEW_PRIVATE_TAB,
-                PRIVATE_BROWSING_PINNED_SHORTCUT -> {
-                    MetricsUtils.Source.SHORTCUT
-                }
-                else -> null
+            store.waitForPIPToClose {
+                processed = openSearch(event,navController,out)
             }
+            return processed
```

Comment:
Would this return `false` if fullscreen media is currently playing and the block of `waitForPIPToClose` wasn't yet executed?
Seeing that in the current implementation `true` is returned if `event != null` and we're already in this case it seems to be okay to return here `true` without the need of the new `processed` property.
We return that this processor will act upon the Intent, just that we now introduce a small delay.

Reply:
Would this return false if fullscreen media is currently playing and the block of waitForPIPToClose wasn't yet executed?
-- No. It should not return false

Seeing that in the current implementation true is returned if event != null and we're already in this case it seems to be okay to return here true without the need of the new processed property.
We return that this processor will act upon the Intent, just that we now introduce a small delay.
-- I understand now that we can return true

```
val event = intent.extras?.getString(HomeActivity.OPEN_TO_SEARCH)
        return if (event != null) {
            store.waitForPIPToClose {
                openSearch(event,navController,out)
            }
            return true
        } else {
            false
        }
```